File: src/main/java/com/github/jacekpoz/common/sendables/Message.java
package com.github.jacekpoz.common.sendables;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;

/**
 * Do I really have to describe this class?
 *
 * @author  jacekpoz
 * @version 0.0.4
 * @since   0.1.0
 */
@ToString
@EqualsAndHashCode
public class Message implements Sendable {

    @Getter
    private final long messageID;
    @Getter
    private final long chatID;
    @Getter
    private final long authorID;
    @Getter @Setter
    private String content;
    @EqualsAndHashCode.Exclude
    @Getter
    private final LocalDateTime dateSent;

    @JsonCreator
    public Message(
            @JsonProperty("messageID") long mID,
            @JsonProperty("chatID") long cID,
            @JsonProperty("authorID") long aID,
            @JsonProperty("content") String text,
            @JsonProperty("dateSent") LocalDateTime send
    ) {
        messageID = mID;
        chatID = cID;
        authorID = aID;
        content = text;
        dateSent = send;
    }

    public Message(String text) {
        this(-1, -1, -1, text, LocalDateTime.MIN);
    }

}

File: src/main/java/com/github/jacekpoz/common/sendables/Message.kt
package com.github.jacekpoz.common.sendables

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

/**
 * Do I really have to describe this class?
 *
 * @author  jacekpoz
 * @since   0.5.0
 */
data class Message @JsonCreator constructor(
    @JsonProperty("messageID") val messageID: Long,
    @JsonProperty("chatID") val chatID: Long,
    @JsonProperty("authorID") val authorID: Long,
    @JsonProperty("content") var content: String,
    @JsonProperty("dateSent") val sentDate: LocalDateTime? = LocalDateTime.MIN,
) : Sendable {

    constructor(
        content: String
    ): this(-1, -1, -1, content, LocalDateTime.MIN)

    @JsonProperty("attachments")
    val attachments: List<Attachment> = ArrayList()

    override fun equals(other: Any?): Boolean = (other is Message) &&
            messageID == other.messageID &&
            chatID == other.chatID &&
            authorID == other.authorID

    override fun hashCode(): Int {
        var result = messageID.hashCode()
        result = 31 * result + chatID.hashCode()
        result = 31 * result + authorID.hashCode()
        return result
    }
}
